explainability-api/microservice/checklist.py

```diff
@@ -0,0 +1,179 @@
+        answer (str) : Prediction for test case made by the skill
+
+    """
+    skill_type = skill["skill_type"]
+    base_model = skill["default_skill_args"].get("base_model")
+    adapter = skill["default_skill_args"].get("adapter")
+    # extract all tests
+    all_tests = [tests["test_cases"] for tests in test_cases]
+    # all_tests = list(itertools.chain.from_iterable([tests["test_cases"] for tests in test_cases]))
+    questions, contexts, answers = list(), list(), list()
+
+    test_type = list(itertools.chain.from_iterable([[test["test_type"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+    capability = list(itertools.chain.from_iterable([[test["capability"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+    test_name = list(itertools.chain.from_iterable([[test["test_name"]] * len(test["test_cases"])
+                                                    for test in test_cases]))
+
+    for tests in all_tests:
+        questions.append([query["question"] for query in tests])
+        # list of list for mcq else list
+        contexts.append([query["context"] if skill_type != "multiple-choice"
+                         else query["context"] + "\n" + "\n".join(query["options"])
+                         for query in tests])
+        answers.extend([query.get("answer") if "answer" in query.keys() else query.get("prediction_before_change")
+                        for query in tests])
+
+        # TODO
+        # send batch to the skill query endpoint
+
+    prediction_requests = list()
+    # create the prediction request
+    for idx in range(len(questions)):
+        for question, context in zip(questions[idx], contexts[idx]):
+            request = dict()
+            request["num_results"] = 1
+            request["user_id"] = "ukp"
+            request["skill_args"] = {"base_model": base_model, "adapter": adapter, "context": context}
+            request["query"] = question
+            prediction_requests.append(request)
+
+    model_inputs = dict()
+    model_inputs["request"] = prediction_requests
+    model_inputs["answers"] = answers
+    model_inputs["test_type"] = test_type
+    model_inputs["capability"] = capability
+    model_inputs["test_name"] = test_name
+    # logger.info("inputs:", model_inputs)
+
+    return model_inputs
+
+
+def predict(model_inputs: dict, skill_id: str) -> list:
+    """
+    Predicts a given query
+
+    Args:
+        model_inputs (dict) : input for the model inference
+        skill_id (str) : id of skill for which the predictions need to be run
+
+    Returns:
+        Returns the model predictions and success rate
+    """
+    model_outputs = list()
+    try:
+        headers = {'Content-type': 'application/json'}
+        skill_query_url = f"https://square.ukp-lab.de/api/skill-manager/skill/{skill_id}/query" #note I hardcoded square URL here
```

Comment:
The url should come from the env variable `SQUARE_API_URL`

Reply:
This part of the code was provided by Haritz. We did not ask him why he hard coded the url but I guess he has some reasons? @HaritzPuerto

Reply:
He's on vacation right now. I think it save to use SQUARE_API_URL. The reason we want make this configurable is to be able to deploy to different environments. BTW, the URL contains the protocol, so something like this should work:
```python
import os
# ...
skill_query_url = f"{os.environ['SQUARE_API_URL']}/skill-manager/skill/{skill_id}/query"
```

Reply:
Okay great, could you please change the code to that? :D Thanks